exchanges/request/request.go

```diff
@@ -264,15 +266,22 @@ func (r *Requester) checkRequest(method, path string, body io.Reader, headers ma
 }
 
 // DoRequest performs a HTTP/HTTPS request with the supplied params
-func (r *Requester) DoRequest(req *http.Request, path string, body io.Reader, result interface{}, authRequest, verbose, httpDebug bool) error {
+func (r *Requester) DoRequest(req *http.Request, path string, body io.Reader, result interface{}, authRequest, verbose, httpDebug, httpRecord bool) error {
 	if verbose {
-		log.Debugf("%s exchange request path: %s requires rate limiter: %v", r.Name, path, r.RequiresRateLimiter())
+		log.Debugf("%s exchange request path: %s requires rate limiter: %v",
+			r.Name,
+			path,
+			r.RequiresRateLimiter())
+
 		for k, d := range req.Header {
 			log.Debugf("%s exchange request header [%s]: %s", r.Name, k, d)
 		}
+
 		log.Debug(body)
 	}
 
+	newbody := fmt.Sprintf("%v", body)
```

Comment:
Can we move this down under the condition check for httpRecord? at this point its unused and an allocation that doesn't need to happen